hack/test-integration.sh

```diff
@@ -37,28 +37,34 @@ KUBE_TEST_API_VERSIONS=${KUBE_TEST_API_VERSIONS:-"v1,extensions/v1beta1;v1,autos
 KUBE_TIMEOUT="-timeout 600s"
 KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=${KUBE_INTEGRATION_TEST_MAX_CONCURRENCY:-"-1"}
 LOG_LEVEL=${LOG_LEVEL:-2}
+KUBE_TEST_ARGS=${KUBE_TEST_ARGS:-}
 
 cleanup() {
+  kube::log::status "Cleaning up etcd"
   kube::etcd::cleanup
   kube::log::status "Integration test cleanup complete"
 }
 
 runTests() {
+  kube::log::status "Starting etcd instance"
   kube::etcd::start
-
   kube::log::status "Running integration test cases"
 
   # TODO: Re-enable race detection when we switch to a thread-safe etcd client
   # KUBE_RACE="-race"
-  KUBE_GOFLAGS="-tags 'integration no-docker' " \
+  KUBE_GOFLAGS="${KUBE_GOFLAGS:-} -tags 'integration no-docker'" \
     KUBE_RACE="" \
     KUBE_TIMEOUT="${KUBE_TIMEOUT}" \
     KUBE_TEST_API_VERSIONS="$1" \
     "${KUBE_ROOT}/hack/test-go.sh" test/integration
 
-  kube::log::status "Running integration test scenario with watch cache on"
-  KUBE_TEST_API_VERSIONS="$1" "${KUBE_OUTPUT_HOSTBIN}/integration" --v=${LOG_LEVEL} \
-    --max-concurrency="${KUBE_INTEGRATION_TEST_MAX_CONCURRENCY}" --watch-cache=true
+  # Run the watch cache tests
+  # KUBE_TEST_ARGS doesn't mean anything to the watch cache test.
+  if [ -z "$KUBE_TEST_ARGS" ]; then
```

Comment:
``${}`` is used all over the place, @ixdy are there other conventions one should be aware of?